src/testlib.ts

```diff
@@ -1,29 +1,80 @@
 'use-strict';
 import * as Ajv from 'ajv';
+import * as path from 'path';
+import * as fs from 'fs';
 
 const ajv = new Ajv({
   allErrors: true,
 });
 
-const SYNC_RESPONSE_SCHEMA = require('../intents/sync.response.schema.json');
-const QUERY_RESPONSE_SCHEMA = require('../intents/query.response.schema.json');
-const EXECUTE_RESPONSE_SCHEMA = require('../intents/execute.response.schema.json');
-const DISCONNECT_RESPONSE_SCHEMA = require('../intents/disconnect.response.schema.json');
-const ON_OFF_STATES_SCHEMA = require('../traits/onoff.states.schema.json');
-const ON_OFF_ATTRIBUTES_SCHEMA = require('../traits/onoff.attributes.schema.json');
-
+/**
+ * Retrieve the JSON schema for the given intent.
+ * @param intentName FQN of the intent
+ * @param schemaType One of 'request' or 'response'
+ */
+function getIntentSchema(intentName: string, schemaType: string): object | undefined {
+  try {
+    const shortIntent = intentName.split('.').pop()?.toLowerCase() || 'unknown';
+    const filename = `${shortIntent}.${schemaType}.schema.json`;
+    return require(path.join('../schema/intents', shortIntent, filename));
```

Comment:
might be safer to use `__dirname` rather than `../` here: https://nodejs.org/docs/latest/api/modules.html#modules_dirname

Reply:
and JSON.parse

Reply:
Path fixed.

`require()` wraps the file read/cache/parse into one operation, and it was also the method previously used to load the schema files (keeping the change small). WDYT?